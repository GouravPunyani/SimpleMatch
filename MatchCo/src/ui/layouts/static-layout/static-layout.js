// @flow

import React, { Component } from 'react'
import { View, StyleSheet } from 'react-native'
import { spacing, color } from '../../theme'

export type Props = {
  children: React.Children,
  style?: ViewStyle
};

export class StaticLayout extends Component <*, Props, *> {
  render () {
    return (
      <View style={[styles.wrapper, this.props.style]}>
        {this.props.children}
      </View>
    )
  }
}

const styles = StyleSheet.create({
  wrapper: {
    flex: 1,
    backgroundColor: color.background,
    justifyContent: 'flex-start',
    alignItems: 'stretch',
    padding: spacing.medium
  }
})
