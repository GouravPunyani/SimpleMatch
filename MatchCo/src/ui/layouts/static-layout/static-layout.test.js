// @flow

import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import { StaticLayout as Comp } from './static-layout'

test('renders correctly', () => {
  const tree = renderer.create(<Comp />)
  expect(tree).toMatchSnapshot()
})
