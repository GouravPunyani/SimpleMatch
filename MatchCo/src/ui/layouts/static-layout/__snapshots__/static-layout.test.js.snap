// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": undefined,
      "flex": 1,
      "justifyContent": "flex-start",
      "padding": 16,
    }
  }
/>
`;
